lib/utils/parameter_utils.dart

```diff
@@ -8,3 +13,133 @@ bool nameConsistsOfUnderscoresOnly(FormalParameter parameter) {
+    'error' => ErrorSeverity.ERROR,
+    'none' => ErrorSeverity.NONE,
+    _ => null,
+  };
+}
+
+/// Extension to create a copy of [LintCode]
+extension LintCodeCopyWith on LintCode {
+  /// Returns a copy of [LintCode] with specified fields replaced
+  LintCode copyWith({
+    String? name,
+    String? problemMessage,
+    String? correctionMessage,
+    String? uniqueName,
+    String? url,
+    ErrorSeverity? errorSeverity,
+  }) =>
+      LintCode(
+        name: name ?? this.name,
+        problemMessage: problemMessage ?? this.problemMessage,
+        correctionMessage: correctionMessage ?? this.correctionMessage,
+        uniqueName: uniqueName ?? this.uniqueName,
+        url: url ?? this.url,
+        errorSeverity: errorSeverity ?? this.errorSeverity,
+      );
+}
+
+/// Source: https://github.com/epam-cross-platform-lab/swagger-dart-code-generator/blob/master/lib/src/extensions/yaml_extensions.dart
+extension YamlMapConverter on YamlMap {
+  dynamic _convertNode(dynamic v) {
+    if (v is YamlMap) {
+      return v.toMap();
+    } else if (v is YamlList) {
+      final list = <dynamic>[];
+      for (final e in v) {
+        list.add(_convertNode(e));
+      }
+      return list;
+    } else {
+      return v;
+    }
+  }
+
+  /// Converts [YamlMap] to [Map<String, dynamic>]
+  Map<String, dynamic> toMap() {
+    final map = <String, dynamic>{};
+    nodes.forEach(
+      (k, v) => map[(k as YamlScalar).value.toString()] = _convertNode(v.value),
+    );
+    return map;
+  }
+}
+
+/// Determines whether the file should be skipped using Glob patterns
+///
+/// See https://pub.dev/packages/glob
+bool shouldSkipFile({
+  required List<String> includeGlobs,
+  required List<String> excludeGlobs,
+  required String path,
+  String? rootPath,
+}) {
+  if (includeGlobs.isEmpty && excludeGlobs.isEmpty) {
+    return false;
+  }
+
+  final bool isIncludeOnly = excludeGlobs.isEmpty && includeGlobs.isNotEmpty;
+  final bool isExcludeOnly = includeGlobs.isEmpty && excludeGlobs.isNotEmpty;
+  final pathNormalized = normalizePath(
+    path,
+    rootPath != null ? normalizePath(rootPath) : null,
+  );
+  if (isIncludeOnly) {
+    return _isFileNotIncluded(includeGlobs, path);
+  }
+
+  if (isExcludeOnly) {
+    return _isFileExcluded(includeGlobs, pathNormalized);
+  }
+
+  final isFileNotIncluded = _isFileNotIncluded(includeGlobs, pathNormalized);
+  final isFileExcluded = _isFileExcluded(excludeGlobs, pathNormalized);
+  return isFileNotIncluded || isFileExcluded;
+}
+
+bool _isFileExcluded(List<String> excludes, String path) {
+  final hasMatch =
+      excludes.map(Glob.new).toList().any((glob) => glob.matches(path));
+  return hasMatch;
+}
+
+bool _isFileIncluded(List<String> includes, String path) {
+  final hasMatch =
+      includes.map(Glob.new).toList().any((glob) => glob.matches(path));
+  return hasMatch;
+}
+
+bool _isFileNotIncluded(List<String> includes, String path) {
+  return !_isFileIncluded(includes, path);
+}
```

Comment:
Could we dedupe a bit?
```suggestion
bool _matchesAnyGlob(List<String> globsList, String path) {
  final hasMatch =
      globsList.map(Glob.new).toList().any((glob) => glob.matches(path));
  return hasMatch;
}

bool _doesNotMatchGlobs(List<String> globList, String path) {
  return !_matchesAnyGlob(globList, path);
}
```